src/bmp/decoder.rs

```diff
@@ -109,17 +109,129 @@ impl<'a> Iterator for RowIterator<'a> {
     }
 }
 
+/// All errors that can occur when attempting to parse a BMP
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum BmpDecoderError {
```

Comment:
Same concerns with instant  `pub` here as in the other PR. The string-errors was essentially the leanest interface possible, allowing arbitrary SemVer compatible additions.

Reply:
Also this is in a module `bmp`, might as well drop the `Bmp`-prefix.

Reply:
Done.

Also of note: this is still perfectly opaque to the user – `DecodingError` stores `Option<Box<dyn Error + Send + Sync>>` which is perfectly unuseful, except to use it as `Display` (this transition should be transparent, too, since all those fields are private).